package/boot/uboot-mvebu/patches/0003-clearfog-enable-distro-boot-code.patch

```diff
@@ -0,0 +1,97 @@
+From 8133a1b22ed1f5c3db3c3facbf04ebe09f086abb Mon Sep 17 00:00:00 2001
+From: Josua Mayer <josua.mayer97@gmail.com>
+Date: Tue, 22 Nov 2016 16:33:23 +0100
+Subject: [PATCH 3/3] clearfog: enable distro boot code
+
+load addresses start at 0x02000000, leaving enough space for fdt,
+boot scripts and ramdisk at the end.
+Only boot from mmc, the other options are not in working order.
+Removed *_high variables in good faith. They can be re-added with sane
+values when, and if required.
+
+Signed-off-by: Josua Mayer <josua.mayer97@gmail.com>
+---
+ Kconfig                    |  1 +
+ include/configs/clearfog.h | 47 +++++++++++++++++++++++++++++++++++++++++-----
+ 2 files changed, 43 insertions(+), 5 deletions(-)
+
+diff --git a/Kconfig b/Kconfig
+index 81b4226463..f7aec2255b 100644
+--- a/Kconfig
++++ b/Kconfig
+@@ -59,6 +59,7 @@ config DISTRO_DEFAULTS
+ 	default y if ARCH_LS2080A
+ 	default y if ARCH_MESON
+ 	default y if ARCH_ROCKCHIP
++	default y if TARGET_CLEARFOG
+ 	default n
+ 	select CMD_BOOTZ if ARM && !ARM64
+ 	select CMD_BOOTI if ARM64
+diff --git a/include/configs/clearfog.h b/include/configs/clearfog.h
+index 0c51d2a288..73dab5d20e 100644
+--- a/include/configs/clearfog.h
++++ b/include/configs/clearfog.h
+@@ -79,11 +79,6 @@
+ 
+ #define CONFIG_SYS_ALT_MEMTEST
+ 
+-/* Keep device tree and initrd in lower memory so the kernel can access them */
+-#define CONFIG_EXTRA_ENV_SETTINGS	\
+-	"fdt_high=0x10000000\0"		\
+-	"initrd_high=0x10000000\0"
+-
+ /* SPL */
+ /*
+  * Select the boot device here
+@@ -128,6 +123,48 @@
+ #endif
+ #endif
+ 
++#ifndef CONFIG_SPL_BUILD
++/*
++ * Add standard bootenv from distro boot code:
++ * Keep device tree and initrd in lower memory so the kernel can access them
++ * Set default load addresses:
++ *  - 63MB space for kernel
++ *  - 1MB space for fdt
++ *  - 1MB space for extlinux file, or boot script
++ *  - remainder for ramdisk
++ * Keep device tree and initrd in lower memory so the kernel can access them
++ * Set name of fdt file
++ * specify console device
++ */
++#define CONFIG_EXTRA_ENV_SETTINGS	\
++	"kernel_addr_r=0x02000000\0" \
++	"fdt_addr_r=0x05c14dc0\0" \
++	"pxefile_addr_r=0x05d09000\0" \
++	"scriptaddr=0x05d09000\0" \
++	"ramdisk_addr_r=0x05dfd240\0" \
```

Comment:
I guess it's not really important then, but it would be nice to have this fixed anyway. I really don't think I will be the only one hit by WTF moments reading those values ;)